intents.go

```diff
@@ -44,6 +46,33 @@ type Stake struct {
 	ChainId string    `json:"chainId"`
 }
 
+type Supply struct {
+	Type AssetType `json:"type,omitempty" binding:"required"`
+
+	// Asset is the contract address for the token to Supply or add into the
+	// Protocol
+	Asset string `json:"asset,omitempty" binding:"required"`
+
+	// can be empty? and solver chooses a default protocol to supply to?
+	// this would be the Contract address for the Protocol
+	// Aave3 as an example would be 0x87870Bca3F3fD6335C3F4ce8392D69350B4fA4E2
+	Protocol string `json:"protocol,omitempty"`
+}
+
+type WithdrawSupply struct {
+	Type AssetType `json:"type,omitempty" binding:"required"`
+	// Asset is the contract address for the token to withdraw or add into the
+	// Protocol
+	Asset string `json:"asset,omitempty" binding:"required"`
+
+	Amount string `json:"amount,omitempty" binding:"required"`
```

Comment:
Why string for amount? I see that we previously used float64, which is limited, but float64 should be the choice for consistency. Ideally, we can also change them all to decimals as long as we coordinate the change with the codebases, i.e., Solver.
1. Strings don't enforce any constraints on the format or precision of the numeric value.
2. It is error-prone. We'd need to parse the strings into actual numeric types first.
3. less efficient in terms of memory usage and performance than using a proper numeric type.

Reply:
I was following existing types we use for addresses and others. I would change that.